### PR TITLE
[eight] Backport the "dynamic trigger" from nine

### PR DESCRIPTION
**Pull request by jaredgrubb**
I had a use-case where this would be very helpful, so I backported a change made in nine by @tardyp that added "dynamic scheduler".

This is mostly the same patch, but I did have to modify it a bit to make it work with eight's scheduler stuff.

Also, I added a note about renderable properties.

### CODE REVIEW

**Comment by tardyp** [comment]
:+1:

File: master/buildbot/steps/trigger.py [hunk 1]
```diff
@@ -80,33 +80,28 @@ def end(self, result):
...
-                    invalid_schedulers.append(scheduler)
-            else:
-                invalid_schedulers.append(scheduler)
-
-        return (triggered_schedulers, invalid_schedulers)
+    def getSchedulerByName(self, name):
+        # Use a quick cache to avoid generating this dict every time.
+        try:
+            all_schedulers = self._all_schedulers
+        except AttributeError:
```
Comment (sa2ajj) [inline master/buildbot/steps/trigger.py]: using exceptions as a flow control?

why not set `self._all_schedulers` to `None` and check against that?

**Comment by jaredgrubb** [comment]
Adjusted the patch to address sa2ajj's comment.

**Comment by Stibbons** [comment]
+1

**Pull request merged**